protob/messages.proto

```diff
@@ -913,6 +938,274 @@ message CosiSignature {
 	optional bytes signature = 1;		// Signature
 }
 
+//////////////////////
+// Stellar messages //
+//////////////////////
+
+/**
+ * Request: Public key at the specified index
+ * @next StellarPublicKey
+ */
+message StellarGetPublicKey {
+	optional uint32 index = 1;		// Account index (derived as m/44'/148'/index')
+}
+
+/**
+ * Response: Public key for the given index
+ * @prev StellarGetPublicKey
+ */
+message StellarPublicKey {
+	optional uint32 index = 1;		// Account index (0-based) used to derive the public key
+	optional bytes public_key = 2;	// Raw bytes of the public key (no version or checksum)
+}
+
+/**
+ * Request: ask device to sign the given string
+ * @next StellarSignedData
+ */
+message StellarSignString {
+	optional uint32 index = 1;		// Account index (0-based)
+
+	optional string message = 2;	// Message to sign
+}
+
+/**
+ * Response: device has signed data
+ * @prev StellarSignString
+ */
+message StellarSignedData {
+	optional uint32 index = 1;		// Account index (0-based) used to derive the signing key
+	optional bytes public_key = 2;	// Raw bytes of the public key (no version or checksum)
+	optional bytes signature = 3;   // ed25519 signature bytes
+}
+
+/**
+ * Request: ask device to verify that signature is valid (public_key has signed message)
+ * @next StellarMessageVerification
+ */
+message StellarVerifyMessage {
+	optional bytes public_key = 1;	// Public key corresponding to the private key that signed the message
+	optional bytes message = 2;		// Binary data that was signed
+	optional bytes signature = 3;	// Signature to verify
+}
+
+/**
+ * Response: device has checked signature and either verified or denied it
+ * @prev StellarVerifyMessage
+ */
+message StellarMessageVerification {
```

Comment:
For consistency, this should be removed and `Success` or `Failure` used instead.